### PR TITLE
Math functions were implemented

### PR DESCRIPTION
**Pull request by angelpilled**
I implemented inverse sine, factorial, exponentiation and square root functions.

### CODE REVIEW

File: kernel/libc/math.c [hunk 1]
```diff
@@ -12,25 +13,69 @@ static double reduce_angle(double angle)
...
+	result *= b;
+	n--;
+    }
+    return result;
+}
+
+// Square root
+double sqrt(double x)
+{
+   return exp(x, 0.5);
```
Comment (moussacodes) [inline kernel/libc/math.c:80]: `exp()` will enter an infinite loop

File: kernel/libc/math.c [hunk 2]
```diff
@@ -12,25 +13,69 @@ static double reduce_angle(double angle)
...
+// Tangent calculation
 double tan(double x)
 {
 	return sin(x)/cos(x);
 }
+
+// Taylor series expansion for inverse sine function (arcsin)
+double arcsin(double x)
+{
+    return x + ( (exp(x, 3))/factorial(3) ) + (( 1 * 3 * exp(x, 5)) / 40.0 ) + ( ( 1 * 3 * 5 * exp(x, 7) ) / 384.0 );
```
Comment (moussacodes) [inline kernel/libc/math.c:43]: replace `factorial(3)` with 6

**Review by moussacodes — CHANGES_REQUESTED** [review]

File: kernel/libc/math.c (hunk 2 above)
Comment (angelpilled) [reply kernel/libc/math.c:43]: Thanks brother, I'm going to fix it right now.

**Comment by angelpilled** [comment]
@moussacodes these should fix the problems.

**Review by lvntky — APPROVED** [review]
checked & approved

**Comment by lvntky** [comment]
thank you guys, merged @moussacodes @angelpilled

File: kernel/libc/math.c [hunk 3]
```diff
@@ -12,25 +13,78 @@ static double reduce_angle(double angle)
...
+}
+
+// Square root
+double sqrt(double x)
+{
+    if (x < 0)
+    {
+	return -1;
+    }
+    return exp(x, 0.5);
```
Comment (moussacodes) [inline kernel/libc/math.c:89]: sqrt should be re-implemented using another approach
[mathematical methods](en.wikipedia.org/wiki/Methods_of_computing_square_roots)
[Newton Raphson Method](https://www.geeksforgeeks.org/newton-raphson-method/)

**Review by moussacodes — CHANGES_REQUESTED** [review]

**Pull request merged**